messageV2.go

```diff
@@ -0,0 +1,300 @@
+package pglogrepl
+
+import (
+	"encoding/binary"
+	"time"
+)
+
+type MessageDecoderV2 interface {
+	MessageDecoder
+	DecodeV2(src []byte, inStream bool) error
+}
+
+type StreamStartMessageV2 struct {
+	baseMessage
+
+	Xid uint32
+	// A value of 1 indicates this is the first stream segment for this XID, 0 for any other stream segment
+	FirstSegment uint8
+}
+
+func (m *StreamStartMessageV2) DecodeV2(src []byte, _ bool) (err error) {
+	if len(src) < 5 {
+		return m.lengthError("StreamStartMessageV2", 5, len(src))
+	}
+
+	var low, used int
+	m.Xid, used = m.decodeUint32(src)
+	low += used
+	m.FirstSegment = src[low]
+
+	m.SetType(MessageTypeStreamStart)
+
+	return nil
+}
+
+type StreamStopMessageV2 struct {
+	baseMessage
+}
+
+func (m *StreamStopMessageV2) DecodeV2(_ []byte, _ bool) (err error) {
+	// stream stop has no data
+	m.SetType(MessageTypeStreamStop)
+
+	return nil
+}
+
+type StreamCommitMessageV2 struct {
+	baseMessage
+
+	Xid               uint32
+	Flags             uint8 // currently unused
+	CommitLSN         LSN
+	TransactionEndLSN LSN
+	CommitTime        time.Time
+}
+
+func (m *StreamCommitMessageV2) DecodeV2(src []byte, _ bool) (err error) {
+	if len(src) < 29 {
+		return m.lengthError("StreamCommitMessageV2", 29, len(src))
+	}
+	var low, used int
+	m.Xid, used = m.decodeUint32(src)
+	low += used
+	m.Flags = src[low]
+	low += 1
+	m.CommitLSN, used = m.decodeLSN(src[low:])
+	low += used
+	m.TransactionEndLSN, used = m.decodeLSN(src[low:])
+	low += used
+	m.CommitTime, _ = m.decodeTime(src[low:])
+
+	m.SetType(MessageTypeStreamCommit)
+
+	return nil
+}
+
+type StreamAbortMessageV2 struct {
+	baseMessage
+
+	Xid    uint32
+	SubXid uint32
```

Comment:
Suggest add spec for SubXid:

    Xid of the subtransaction (will be same as xid of the transaction for top-level transactions).